contracts/examples/SwapExamples.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import '../libraries/TransferHelper.sol';
+import '../interfaces/ISwapRouter.sol';
+
+contract SwapExamples {
+    //// For the scope of these swap examples,
+    //// we will detail the design considerations when using
+    //// `exactInput`, `exactInputSingle`, `exactOutput`, and  `exactOutputSingle`.
+
+    //// It should be noted that for the sake of these examples, we purposefully pass in the swap router instead of inherit the swap router for simplicity.
+    //// More advanced example contracts will detail how to inherit the swap router safely.
+
+    ISwapRouter public immutable swapRouter;
+
+    //// This example swaps DAI/WETH9 for single path swaps and DAI/USDC/WETH9 for multi path swaps.
+
+    address constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+
+    //// For this example, we will set the pool fee to 0.3%.
+    uint24 constant poolFee = 3000;
+
+    constructor(ISwapRouter _swapRouter) {
+        swapRouter = _swapRouter;
+    }
+
+    /// @notice swapInputSingle swaps DAI for WETH9 using the DAI/WETH9 0.3% pool by calling `exactInputSingle` in the swap router.
+    /// @param amountIn The exact amount of DAI that will be swapped for WETH9.
+    /// @return amountOut The amount of WETH9 received.
+    function swapInputSingle(uint256 amountIn) external returns (uint256 amountOut) {
+        //// Transfer the specified amount of DAI to this contract.
+        TransferHelper.safeTransfer(DAI, address(this), amountIn);
+
```

Comment:
I think Noah just means that in the docs somewhere we should make a note that because this contract transfers funds from msg.sender, the caller needs to approve this contract

Reply:
If there are any approves needed to make the example work we should include them in the code imo

Reply:
follow up - EOA needs to send an independent approve call before the contract function - my b